### ISSUE TITLE
Fix false positives regression for mixed-case ignoreProperties in value-keyword-case

### ISSUE THREAD

**Issue by camiblanch** [stylelint/stylelint#4644]
> Clearly describe the bug

When I try to use `ignoreProperties` with `value-keyword-case`, it doesn't ignore scss variables with mixed casing. I tried using various variable names and nothing seemed to stop the bug from happening without changing the case. 

I also tested to see if I kept the casing the same in my css file, but changed the casing of my stylelint rule to just be all lowercase. That worked. I found this to be true using strings or regular expressions.

> Which rule, if any, is the bug related to?

`value-keyword-case`

> What code is needed to reproduce the bug?

```css
$fontFamily: Menlo, Monaco, Consolas, monospace;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "value-keyword-case": [
      "lower",
      {"ignoreProperties": ["$fontFamily"]}
    ]
  }
}
```

> Which version of stylelint are you using?

`13.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.css"`
Also found true on the [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIAkAzCA7ALgMQIYFsBLAGwE94ACAWRnWIgBpqNdImBhDAZwmNy6b4MELgAdWMANwAddCAYgupLLgAeCcFy7zwGFIQDmG4LIoVpIAE4BXYjC4XKJ9GbMWAbrmLWYAWgDWMKQA7hCWUL5g-DCOFADapq5uIPTBMJYWDIlJJiCG6GEwAAqWEKLpmIT2sQnIaFh4RGQWALoAvtkULYkd6G0gbUA)

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it uses SCSS variables

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
styles/text.css
 27:37  ✖  Expected "Menlo" to be "menlo"         value-keyword-case
 27:72  ✖  Expected "Monaco" to be "monaco"       value-keyword-case
 27:80  ✖  Expected "Consolas" to be "consolas"   value-keyword-case
```

**Comment by jeddy3** [comment]
@camiblanch Thanks for the clearly written report and for using the template.

I think you're right, this is a regression [introduced in #4611](https://github.com/stylelint/stylelint/pull/4611#issuecomment-597905525).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
